cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractKafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -0,0 +1,230 @@
+            .endMetadata()
+            .withNewSpec()
+            .withNewKafka()
+            .withListeners(new GenericKafkaListenerBuilder()
+                    .withName("plain")
+                    .withPort(9092)
+                    .withType(KafkaListenerType.INTERNAL)
+                    .withTls(false)
+                    .build())
+            .endKafka()
+            .withNewCruiseControl()
+            .endCruiseControl()
+            .endSpec()
+            .build();
+    protected static final KafkaRebalanceSpec EMPTY_KAFKA_REBALANCE_SPEC = new KafkaRebalanceSpecBuilder().build();
+    protected static final PlatformFeaturesAvailability PFA = new PlatformFeaturesAvailability(false, KubernetesVersion.MINIMAL_SUPPORTED_VERSION);
+    protected static KubernetesClient client;
+    protected static MockKube3 mockKube;
+    protected static Vertx vertx;
+    protected static WorkerExecutor sharedWorkerExecutor;
+    protected String namespace;
+    protected KafkaRebalanceAssemblyOperator krao;
+    protected ResourceOperatorSupplier supplier;
+
+    protected static int cruiseControlPort;
+    protected static File tlsKeyFile;
+    protected static File tlsCrtFile;
+    protected static MockCruiseControl cruiseControlServer;
+
+    @BeforeAll
+    public void beforeAll() throws IOException {
+        // Configure the Kubernetes Mock
+        mockKube = new MockKube3.MockKube3Builder()
+                .withKafkaCrd()
+                .withKafkaRebalanceCrd()
+                .withDeletionController()
+                .build();
+        mockKube.start();
+        client = mockKube.client();
+
+        vertx = Vertx.vertx();
+        sharedWorkerExecutor = vertx.createSharedWorkerExecutor("kubernetes-ops-pool");
+
+        // Configure Cruise Control mock
+        cruiseControlPort = TestUtils.getFreePort();
+        tlsKeyFile = ReadWriteUtils.tempFile(KafkaRebalanceAssemblyOperatorTest.class.getSimpleName(), ".key");
+        tlsCrtFile = ReadWriteUtils.tempFile(KafkaRebalanceAssemblyOperatorTest.class.getSimpleName(), ".crt");
+
+        new MockCertManager().generateSelfSignedCert(tlsKeyFile, tlsCrtFile,
+                new Subject.Builder().withCommonName("Trusted Test CA").build(), 365);
+
+        cruiseControlServer = new MockCruiseControl(cruiseControlPort, tlsKeyFile, tlsCrtFile);
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        sharedWorkerExecutor.close();
+        vertx.close();
+        mockKube.stop();
+        if (cruiseControlServer != null && cruiseControlServer.isRunning()) {
+            cruiseControlServer.stop();
+        }
+    }
+
+    @BeforeEach
+    public void beforeEach(TestInfo testInfo) {
+        namespace = testInfo.getTestMethod().orElseThrow().getName().toLowerCase(Locale.ROOT);
+        // handling Kubernetes constraint for namespace limit name (63 characters)
+        if (namespace.length() > 63) {
+            namespace = namespace.substring(0, 63);
+        }
```

Comment:
Other test classes don't seem to have this check, is it needed?

Reply:
https://github.com/strimzi/strimzi-kafka-operator/pull/11348#discussion_r2089293297